extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorpvault/HashicorpCertificateResolver.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2022 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security.hashicorpvault;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.security.CertificateResolver;
+import org.eclipse.dataspaceconnector.spi.security.Vault;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+/**
+ * Resolves an X.509 certificate in Hashicorp vault.
+ */
+public class HashicorpCertificateResolver implements CertificateResolver {
```

Comment:
It would probably be best to move the PEMUtil upwards and use it in both modules. Alternatively I can use the Azure-based method here too. Which would you prefer?

Reply:
Hi @bcronin90 I would put PEMUtils in a library or abstract out the Certificate Resolver. @ndr-brt and @paullatzelsperger what do you think?

Reply:
I'd abstract `CertificateResolver` to have an unique instance like the `AzureCertificateResolver` that relies only on base java libraries (does not require bouncycastle), could be called `DefaultCertificateResolver` and be registered as default implementation for `CertificateResolver` and avoid `NoopCertificateResolver` at all.

Reply:
Is the idea to have the bouncycastle dependency isolated to PEMUtil or to remove it entirely?

Reply:
@bcronin90 in my opinion for this task is unnecessary. The `AzureCertificateResolver` is not using it, it uses only base java classes.

Reply:
I can rewrite the `HashicorpCertificateResolver` to use `CertificateFactory` like `AzureCertificateResolver` does. But I'd really prefer it the other way around. I think the BouncyCastle implementation is a bit less clunky, especially because it doesn't rely on manually replacing parts of a string. I'm open to either direction though.

Reply:
I would prefer keeping the bouncycastle dependency. It appears, other packages already use it, so it's not a new dependency.

Reply:
Could you have a look at this @juliapampus ? I completely agree with the rest of the review by @ndr-brt , but this detail is a bit problematic. I tried to sum it up in the above comment, but if you need any further info, please ask.

Reply:
I agree with @ndr-brt - we should not use Bouncycastle and in general, should avoid unnecessary dependencies. I don't think the Azure approach is kludgey since the certificate format is standard.

Reply:
Also, we should support a common format across Azure Vault and Hashicorp

Reply:
Thanks for the input @jimmarino . I've removed the bouncycastle dependency and consolidated the certificate validation. I'll push it with the rest of the review implementation. I do however have some difficulty with removing bouncycastle as a test dependency. We rely on generated certificates for our tests and creating those with plain java is an extreme pain. So my proposition would be to keep the test dependency, as it isn't deployed with the project and thus doesn't bloat it.

Reply:
I don't think we should have the dependency. Please remove it. The creation of test certificates can be encapsulated with a test utility method that is shared across modules if needed.

Reply:
I've already removed it.